libs/vkd3d/meta_commands.c

```diff
@@ -20,6 +20,121 @@
 
 #include "vkd3d_private.h"
 
+const GUID IID_META_COMMAND_DSTORAGE = {0x1bddd090,0xc47e,0x459c,{0x8f,0x81,0x42,0xc9,0xf9,0x7a,0x53,0x08}};
+
+struct d3d12_meta_command_dstorage_create_args
+{
+    UINT64 version;
+    UINT64 format;
+    UINT64 max_streams;
+    UINT64 flags;
+};
+
+struct d3d12_meta_command_dstorage_exec_args
+{
+    D3D12_GPU_VIRTUAL_ADDRESS input_buffer_va;
+    UINT64 input_buffer_size;
+    D3D12_GPU_VIRTUAL_ADDRESS output_buffer_va;
+    UINT64 output_buffer_size;
+    D3D12_GPU_VIRTUAL_ADDRESS control_buffer_va;
+    UINT64 control_buffer_size;
+    D3D12_GPU_VIRTUAL_ADDRESS scratch_buffer_va;
+    UINT64 scratch_buffer_size;
+    UINT64 stream_count;
+    D3D12_GPU_VIRTUAL_ADDRESS status_buffer_va;
+    UINT64 status_buffer_size;
+};
+
+struct d3d12_meta_command_parameter_info
+{
+    D3D12_META_COMMAND_PARAMETER_STAGE stage;
+    D3D12_META_COMMAND_PARAMETER_DESC desc;
+};
+
+struct d3d12_meta_command_info
+{
+    REFGUID command_id;
+    LPCWSTR name;
+    D3D12_GRAPHICS_STATES init_dirty_states;
+    D3D12_GRAPHICS_STATES exec_dirty_states;
+    unsigned int parameter_count;
+    const struct d3d12_meta_command_parameter_info *parameters;
+    d3d12_meta_command_create_proc create_proc;
+};
+
+static const struct d3d12_meta_command_parameter_info d3d12_meta_command_dstorage_parameter_infos[] =
```

Comment:
How was this determined? By enumerating parameters from a native implementation?

Reply:
Yes, these match the output of EnumerateMetaCommandParameters.